source/auth/auth.md

```diff
@@ -2044,6 +2045,8 @@ to EC2 instance metadata in ECS, for security reasons, Amazon states it's best p
 
 ## Changelog
 
+- 2024-04-24: Clarify that TOKEN_RESOURCE for MONGODB-OIDC must be url-encoded.
```

Comment:
This reminds me, should we url-encode the resource when sending to GCP? For example, I currently have,

```
    public static CredentialInfo fetchGcpCredentialInfo(final String resource) {
        String endpoint = "http://metadata/computeMetadata/v1/instance/service-accounts/default/identity?audience="
                + resource;
```
but I think we want to specify that we should wrap `resource` in `urlEncode(resource)`?

Reply:
It occurs to me that if we are not doing this, then users can just pass in url-encoded values to TOKEN_RESOURCE, which would then remain unprocessed, and appended here - so no spec changes would be needed?

Reply:
The idea was to specify what value we are expecting there. If it should be plain value - then driver should urlEncode it while building the GCP or Azure callback Url. If we expect encoded value - then we have to validate if it was encoded properly and just concatenate it without url encoding. But @blink1073 decided to have encoding auto-detection functionality which might be useful, but looks confusing at least for me.

Reply:
My point was - users would most likely *have* to encode when using a connection string, but they won't have to when using MongoClient input.  I'm saying use whatever mechanism makes sense to ensure it is url-encoded without double-encoding.

Reply:
If we just let the value though it might be a problem, for example if user set `TOKEN_RESOURCE = "token&another-magic-parameter=42"`.

Reply:
Then that is a malformed connection string

Reply:
I updated it to be more clear but less prescriptive on the how.

Reply:
If you creating mongo client from connection string - yes. But if you creating mongo client by constructing it from properties dictionary. Should the value be encoded or not?

Reply:
I think we should allow either, so it is easier to copy-paste between contexts (like from a connection string to a dictionary).

Reply:
In other comments I argued that it is impossible to detect whether a string is encoded or not, so I believe the only viable option is to expect only raw unencoded values when constructing "in code" (via dictionary, etc.) rather than from the connection string.

(Others, feel free to resolve this convo, might be clearer to finish remaining discussions elsewhere.)